include/mqt-core/dd/Operations.hpp

```diff
@@ -186,17 +141,17 @@ getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
 // An empty permutation marks the identity permutation.
 
 template <class Config>
-qc::MatrixDD getDD(const qc::Operation* op, Package<Config>& dd,
+qc::MatrixDD getDD(const qc::Operation& op, Package<Config>& dd,
```

Comment:
Given the comment above, I am also thinking whether it wouldn't be cleaner to separate the `inverse` function from the `non-inverse` function here.
I think the duplication would be at a minimum and the code would be easier to read overall. 
What do you think?